File: old_groovy/src/main/groovy/net/zomis/brainf/model/ast/tree/Syntax.java
package net.zomis.brainf.model.ast.tree;

import net.zomis.brainf.model.BrainfuckRunner;
import net.zomis.brainf.model.ast.Token;

import java.util.ArrayList;
import java.util.List;

public abstract class Syntax {

    private final List<Token> tokens = new ArrayList<>();

    public List<Token> getTokens() {
        return tokens;
    }

    public abstract void perform(BrainfuckRunner runner);

}

File: brainduck-kotlin/src/commonMain/kotlin/net/zomis/brainf/model/ast/tree/Syntax.kt
package net.zomis.brainf.model.ast.tree;

import net.zomis.brainf.model.BrainfuckRunner
import net.zomis.brainf.model.ast.Token

abstract class Syntax {

    val _tokens = mutableListOf<Token?>()
    val tokens = _tokens.toList()

    abstract fun perform(runner: BrainfuckRunner)

}
